File: B. Ordinary Numbers.py

def Soluiton():
    N=int(input())
    ans=0
    for pw in  range(1,N+1):
        for d in range(1,10):
            if(pw*d <=N):
                ans+=1
        pw= pw*10
    return ans



def main():
    
    test=int(input())
    for _ in range(test):
        print(Soluiton())
    
    
    
    
    
if __name__=="__main__":
    main()

File: B. Ordinary Numbers.cpp
#include<bits/stdc++.h>
using namespace std;

using ll=long long;

void Solution()
{
    ll N;
    cin>>N;
    int ans=0;

    for(ll pw=1;pw<=N;pw=pw*10+1)
    {
        for(int d=1;d<=9;d++)
        {
            if(d*pw<=N)
                ans++;
        }
    }
    cout<<ans<<endl;

}


int main()
{
    int test;
    cin>>test;

    while(test--)
    {
        Solution();
    }

    return 0;
}
